bluesky/bundlers.py

```diff
@@ -660,8 +691,39 @@ async def collect(self, msg):
 
             descriptor_uid = doc["uid"]
             local_descriptors[frozenset(stream_data_keys)] = (stream_name, descriptor_uid)
+        self._local_descriptors[collect_obj] = local_descriptors
+
+    async def collect(self, msg):
+        """
+        Collect data cached by a flyer and emit documents.
+
+        Expect message object is
+
+            Msg('collect', collect_obj)
+            Msg('collect', flyer_object, stream=True, return_payload=False)
+        """
+        collect_obj = check_supports(msg.obj, Flyable)
+
+        if not self.run_is_open:
+            # sanity check -- 'kickoff' should catch this and make this
+            # code path impossible
+            raise IllegalMessageSequence(
+                "A 'collect' message was sent but no run is open."
+            )
+        self._uncollected.discard(collect_obj)
 
-            bulk_data[descriptor_uid] = []
+        # Resource and Datum documents
+        for name, doc in maybe_collect_asset_docs(collect_obj):
+            # Add a 'run_start' field to the resource document on its way out.
+            if name == "resource":
+                doc["run_start"] = self._run_start_uid
+            await self.emit(DocumentNames(name), doc)
+
+        # Populate descriptors and local descriptors from describe_collect
+        # if not already called
+        if collect_obj not in self._local_descriptors:
+            await self._cache_describe_collect(collect_obj)
```

Comment:
This means `describe_collect` will only be called once per run for `Flyable`, like `describe` is for `Readable`. Is this correct?

Reply:
Note that it used to be called every time we called `collect()`

Reply:
[dallan (NSLS-II)](https://app.slack.com/team/U754JQZAM)  [1:33 PM](https://nikea.slack.com/archives/D01A8S4QTU0/p1648557235141449)
No, that should be called once (like describe()). I suspect that is a hold-over from when collect() was only ever called once, before support for partial, incremental collect was added.